tests/run-make/compiler-lookup-paths/rmake.rs

```diff
@@ -0,0 +1,103 @@
+// Since #19941, rustc can accept specifications on its library search paths.
+// This test runs Rust programs with varied library dependencies, expecting them
+// to succeed or fail depending on the situation.
+// The second part of the tests also checks that libraries with an incorrect hash
+// fail to be used by the compiler.
+// See https://github.com/rust-lang/rust/pull/19941
+
+use run_make_support::fs_wrapper;
+use run_make_support::{rmake_out_path, rustc};
+
+fn main() {
+    assert!(rmake_out_path("libnative.a").exists());
```

Comment:
Note: I don't think this assert is at the right place (checking for output artifact before it exists)

Reply:
The Makefile starts with `all: $(TMPDIR)/libnative.a`. I thought this meant "this file is a dependency, it must exist before proceeding."

Reply:
It is already starting to assume libnative.a exists before any rustc calls:

`mv $(TMPDIR)/libnative.a $(TMPDIR)/native`

Which I find strange - the current CI fail says libnative.a could not be found. Where does it even come from?

Reply:
I think it's `tools.mk` auto-compiling any existing `.c` files in the directory.

Reply:
```
$(TMPDIR)/lib%.o: %.c
	$(call COMPILE_OBJ,$@,$<)
```

Cursed. Alright, thanks for finding this!

Reply:
> I think it's `tools.mk` auto-compiling any existing `.c` files in the directory.

Oh. OH.